src/scripts/Joyride.jsx

```diff
@@ -268,6 +269,18 @@ class Joyride extends React.Component {
     }
 
     if (nextState.index !== index && isRunning) {
+      if (nextStep && !this.getStepTargetElement(nextStep)) {
+        console.warn('Target not mounted', nextStep, action); //eslint-disable-line no-console
+        this.triggerCallback({
+          action,
+          index,
+          type: callbackTypes.TARGET_NOT_FOUND,
+          step
+        });
+        this.stop();
```

Comment:
I had forgotten that this part of code is inside the `componentWillUpdate`, during which you cannot call `setState`, so stopping does not work. We'll have to do it somewhere else.